server/src/domain/library/library.service.ts

```diff
@@ -162,18 +175,30 @@ export class LibraryService extends EventEmitter {
 
   async unwatch(id: string) {
     if (this.watchers[id]) {
-      await this.watchers[id]();
+      this.watchers[id]();
```

Comment:
Shouldn't this be awaited?

Reply:
according to vscode: 'await' has no effect on the type of this expression.ts(80007)

It doesn't seem like await is needed after the refactor, and my testing makes me think I agree. Do you? It's no longer a record of async functions

Reply:
I looked at this and `watch` returns `return () => watcher.close();`, where `watcher.close()` returns `Promise<void>`. You aren't awaiting that promise if you remove the `await` keyword here.

Reply:
I dug deeper and changed the storage repository to actually return a promise, which it didn't do before. Now it's being awaited correctly!

Reply:
Nice!